### PR TITLE
useSubscribe options

### PR DESCRIPTION
**Pull request by timoxley**
Breaking change.

* Makes` useSubscribe` take an options object for `onMessage`, `onError`, instead of positional parameters.
* Adds two additional event handlers: `onSubscribed`, `onUnsubscribed`. These are used in platform.
* Adds a flag `isActive` so it's possible to control when subscription occurs.
* Updates minimum client version to get browser signing fixes.

Update:
* Adds `onResent` option
* Adds `isRealtime=true` flag for toggling between `subscribe`/`resend` methods
* Hooks up subscription `'error'` event to `onError`
* Uses `streamr-client` v5 async subscription

### CODE REVIEW

**Comment by timoxley** [comment]
Would be good to get some tests on this

File: src/useSubscription.ts [hunk 1]
```diff
@@ -115,11 +143,35 @@ const useSubscription = (subscriptionParams: object, onMessage: (message: object
...
+    }, [client, params, isActive, isRealtime, resubscribeCount])
+
+    useEffect(() => {
+        if (!sub || !isActive) { return }
+        onSubscribed()
+    }, [sub, isActive, onSubscribed])
+
+    useEffect(() => () => {
+        if (sub) {
+            client.unsubscribe(sub!)
```
Comment (mondoreale) [inline src/useSubscription.ts]: The usage of [`!`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) in TS is interesting. However, it is very strange that TS cannot recognize that in this context `sub` is not falsy. Does it really complain?

File: src/useSubscription.ts [hunk 2]
```diff
@@ -43,19 +44,40 @@ const reducer = (state: State, action: Action): State => {
...
+    } = options
 
     const onMessageRef = useRef(onMessage)
 
     useEffect(() => {
         onMessageRef.current = onMessage
     }, [onMessage])
 
-    const onErrorRef = useRef(onError || defaultErrorHandler)
+    const onErrorRef = useRef(onError)
```
Comment (mondoreale) [inline src/useSubscription.ts:77]: 👍

**Review by mondoreale — APPROVED** [review]
Right, testing is the missing part. There's already a ticket for it in Linear.

PR looks good to me! 👍

File: src/useSubscription.ts (hunk 1 above)
Comment (timoxley) [reply src/useSubscription.ts]: yeah I was confused as to why it didn't work, the conditional was right there, however on removing the `!` I get no warnings now 🎉 , so I just removed it.
The `!` was added when the code looked a little different, so maybe something about the former structure confused it? idk

**Pull request merged**